src/main/jbake/content/blog/2017/08/opentracing-appmetrics-canary.adoc

```diff
@@ -0,0 +1,157 @@
+
+* updating the versions in the `pom.xml` files, within the `simple/accountmgr` and `simple/ordermgr` folders
+from `0.0.1` to `0.0.2`
+* re-run the `mvn clean install docker:build` command from the `simple` folder
+* deploy the canary versions of the services using the command `oc create -f services-canary-kubernetes.yml`
+
+As our services `accountmgr` and `ordermgr` determine the backing deployment based on the respective labels
+`app: accountmgr` and `app: ordermgr`, simply having a second deployment with these labels will make them serve requests
+in a round-robin manner.
+
+This deployment script has been pre-configured with the `0.0.2` version, and to only start a single instance of the
+new version of the services. This may be desirable if wanting to monitor the behaviour of the new service versions over
+a reasonable time period, but as we want to see results faster we will scale them up to see more activity. This is
+achieved by expanding the deployment area for each service in the OpenShift console, and selecting the up arrow to
+scale up each service:
+
+ifndef::env-github[]
+image::/img/blog/2017/2017-08-18-canary-scale-up.png[caption="Figure 1: ", title="Scaling up canary deployment"]
+endif::[]
+ifdef::env-github[]
+image::../../../../assets/img/blog/2017/2017-08-18-canary-scale-up.png[caption="Figure 1: ", title="Scaling up canary deployment"]
+endif::[]
+
+Now we can monitor the Prometheus dashboard, using the following query, to see the error ratio per service and version:
+
+```
+sum(increase(span_count{error="true",span_kind="server"}[1m])) without (pod,instance,job,namespace,endpoint,transaction,error,operation,span_kind) / sum(increase(span_count{span_kind="server"}[1m])) without (pod,instance,job,namespace,endpoint,transaction,error,operation,span_kind)
+```
+
+This result of this query can be seen in the image at the top of the article. This chart shows that version `0.0.2` of the
+`accountmgr` service has not generated any errors, while the `0.0.2` of the `ordermgr` appears to be less error prone that
+version `0.0.1`.
+
+Based on these results we may conclude that these versions are better, or atleast no worse, than the previous version - and
+so we should update the main service deployments to use the new version. In the OpenShift UI this can be achieved by
+selecting the three vertical dots on the right hand side of the deployment region, and select the "Edit Yaml" menu item. This
+will display an editor window where the version can be changed from `0.0.1` to `0.0.2`.
+
+
+ifndef::env-github[]
+image::/img/blog/2017/2017-08-18-canary-service-update.png[caption="Figure 2: ", title="Update the service version"]
+endif::[]
+ifdef::env-github[]
+image::../../../../assets/img/blog/2017/2017-08-18-canary-service-update.png[caption="Figure 2: ", title="Update the service version"]
+endif::[]
+
+Once this is saved, the OpenShift UI will show that the service is going through a "rolling update", to incrementally change
```

Comment:
"After you save the YAML configuration file, in the web console you can see the service going through a "rolling update" as OpenShift incrementally changes each service instance over to the new version.